core/drivers/ls_sfp.c

```diff
@@ -0,0 +1,423 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022 Microsoft
+ *
+ * Driver for the NXP LX2160A-series Security Fuse Processor (SFP).
+ */
+
+#include <assert.h>
+#include <drivers/ls_sfp.h>
+#include <io.h>
+#include <kernel/boot.h>
+#include <kernel/dt.h>
+#include <kernel/tee_time.h>
+#include <libfdt.h>
+#include <mm/core_memprot.h>
+#include <util.h>
+
+/**
+ * struct ls_sfp_registers - Memory map of the SFP registers.
+ * RSVD0[0x4]:		Reserved.
+ * @ingr:		Instruction Register.
+ * @svhesr:		Secret Value Hamming Error Status Registers.
+ * @sfpcr:		SFP Configuration Register.
+ * @RSVD1[0x3]:		Reserved.
+ * @version:		SFP Version Register.
+ * @RSVD2[0x71]:	Reserved.
+ * @ospr0:		OEM Security Policy Register 0.
+ * @ospr1:		OEM Security Policy Register 1.
+ * @dcvr0:		Debug Challenge Value Register 0.
+ * @dcvr1:		Debug Challenge Value Register 1.
+ * @drvr0:		Debug Response Value Register 0.
+ * @drvr1:		Debug Response Value Register 1
+ * @fswpr:		Factory Section Write Protect Register.
+ * @fuidr0:		Factory Unique ID Register 0.
+ * @fuidr1:		Factory Unique ID Register 1.
+ * @isbccr:		ISBC Configuration Register.
+ * @fspfr[0x3]:		Factory Scratch Pad Fuse Registers.
+ * @otpmkr[0x8]:	One Time Programmable Master Key.
+ * @srkhr[0x8]:		Super Root Key Hash.
+ * @ouidr[0x5]:		OEM Unique ID Scratch Pad Fuse Registers.
+ */
+static struct ls_sfp_registers {
+	uint32_t RSVD0[0x4];	/* 0x000 */
```

Comment:
Since `ingr` is starting from `0x20`, so first reserved should be `uint32_t RSVD0[0xA];`

Reply:
Shouldn't this actually be `0x20 / 4 = 0x8`?